### DISCUSSION TITLE
Is there a known issue that can cause the Mac Frame counter and the Key Index in Auxiliary security Header go to zero?

### DISCUSSION THREAD

**Discussion by jinpeng1989** [openthread/openthread#8842] [Q&A]
Is there a known issue that can cause the Mac Frame counter and the Key Index in Auxiliary security Header go to zero?
![key_index](https://user-images.githubusercontent.com/97089879/223430898-4237f148-0088-41a8-85d8-5a1f44208a3c.JPG)

**Comment by jinpeng1989** [comment]
The Mac Frame counter and the Key Index will suddenly go to zero and then go back to normal. Is there any way to locate the issue?
![image](https://user-images.githubusercontent.com/97089879/223443654-f9391481-454e-4476-980f-847584fed905.png)

**Comment by jwhui** [comment]
I'm not aware of any platform-independent issues that would cause the frame counter to be set occasionally to 0.

In general, the radio driver is responsible for setting the MAC Frame Counter. Here's one example:
https://github.com/openthread/ot-nrf528xx/blob/886e7f200512f2f9e1e13dfaa42e335ad8258d24/src/src/radio.c#L298

Reach out to your radio vendor to see if they can provide any support.

**Comment by rretanubun** [comment]
Hello,

I am also experiencing this issue, the effect for me is that some packets sent between an `FTD -> MTD` can be missed.

In my setup I am sending pings 1 per second from `PC -> TBR -> TME -> MTD` and experience 2 packets out of 55K are missing and one of the missing ping is time correlated to when the TME -> MTD sends a frame as described in the packet capture.

I also notice that the MAC Frame Counter (`txAckProcessSecurity`) is keyed to `#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2`

I am wondering if the feature is expected to work if some devices (i.e. the MTD) is on an older Thread version `OT_THREAD_VERSION_1_1` or is this purely a function of the frame sender (the rest of the devices in my setup is `OT_THREAD_VERSION_1_3`).

**Reply by jwhui** [reply]
As mentioned in https://github.com/openthread/openthread/discussions/8842#discussioncomment-5233886, the Key ID and Frame Counter values are set by the radio driver.

> I am wondering if the feature is expected to work if some devices (i.e. the MTD) is on an older Thread version OT_THREAD_VERSION_1_1 or is this purely a function of the frame sender (the rest of the devices in my setup is OT_THREAD_VERSION_1_3).

Setting the Key ID and Frame Counter only occurs on the transmitter. Whether there are neighboring devices with older or different versions should not affect the transmit behavior.
